### PR TITLE
Add local docker test script

### PR DESCRIPTION
**Pull request by CjS77**
Have you ever wished you could run tari tests in Ubuntu on OSX but had no
idea how to do it? Well now you can! If you have Docker installed,
this script will pull a docker image (with the required libraries
precompiled), copy your local source over,
and run the tests.

### CODE REVIEW

File: scripts/test_in_docker.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+echo "Creating docker container.."
+# sleep infinity is used to keep the container alive forever
+docker run -d --name $CONTAINER $IMAGE /bin/sleep infinity
+echo "Container is ready"
+
+echo "Copying source files..."
+# The simplest and laziest approach to not copying 6+GB of irrelevant files to the container
+cargo clean
+# We could pull this from git, but this lets you test local changes
+docker cp . $CONTAINER:/src/
```
Comment (sdbondi) [inline scripts/test_in_docker.sh]: One option could be to mount the local dir into the docker image for testing.
This is the one liner I use
` docker run --rm -v`pwd`:/app -it quay.io/tarilabs/rust_tari-build-with-deps:nightly-2019-10-04 /bin/bash`

Comment (CjS77) [reply scripts/test_in_docker.sh]: You'll then get a polluted `target` folder anyway because they'll be filled with linux binaries

Comment (CjS77) [reply scripts/test_in_docker.sh]: `.dockerignore` would be ideal, except it looks like it's ignored in the cp command

Comment (sdbondi) [reply scripts/test_in_docker.sh]: Could be - I suspect they don't hash to the same file name, so mac/linux binaries won't interfere with each other - haven't tested this though.

Comment (CjS77) [reply scripts/test_in_docker.sh]: could be.. lemme test this quick

Comment (CjS77) [reply scripts/test_in_docker.sh]: You sold me.

**Pull request merged**